doc/api/diagnostics_channel.md

```diff
@@ -407,6 +457,526 @@ channel.subscribe(onMessage);
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (message) => {
+  return { message };
+});
+```
+
+#### `channel.unbindStore(store)`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `store` {AsyncLocalStorage} The store to unbind from the channel.
+* Returns: {boolean} `true` if the store was found, `false` otherwise.
+
+Remove a message handler previously registered to this channel with
+[`channel.bindStore(store)`][].
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store);
+channel.unbindStore(store);
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+const { AsyncLocalStorage } = require('node:async_hooks');
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store);
+channel.unbindStore(store);
+```
+
+#### `channel.runStores(data, fn[, thisArg[, ...args]])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `data` {any} Message to send to subscribers and bind to stores
+* `fn` {Function} Handler to run within the entered storage context
+* `thisArg` {any} The receiver to be used for the function call.
+* `...args` {any} Optional arguments to pass to the function.
+
+Publishes data to the channel and applies it to any AsyncLocalStorage instances
+bound to the channel for the duration of the given function. If a transform
+function was given to [`channel.bindStore(store)`][] it will be applied to
+transform the message data before it becomes to context value for the store.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (message) => {
+  const parent = store.getStore();
+  return new Span(message, parent);
+});
+channel.runStores({ some: 'message' }, () => {
+  store.getStore(); // Span({ some: 'message' })
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+const { AsyncLocalStorage } = require('node:async_hooks');
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (message) => {
+  const parent = store.getStore();
+  return new Span(message, parent);
+});
+channel.runStores({ some: 'message' }, () => {
+  store.getStore(); // Span({ some: 'message' })
+});
+```
+
+### Class: `TracingChannel`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+The class `TracingChannel` is a collection of [TracingChannel Channels][] which
+together express a trace. It is used to formalize and simplify the process of
```

Comment:
```suggestion
together express a span. It is used to formalize and simplify the process of
```

Or maybe `operation`. The term trace is usually a tree of linked spans/operations and a single tracing channel forms one entry of a trace not the whole trace.

Reply:
I don't think "span" or "operation" are great terms here either as there's zero agreement on terminology for what those units are called between APMs. It also means additional terminology that likely needs more explaining. This feature is not intended exclusively for APM and what can be considered a "trace" can be a single unit just as much as a collection of units. I think given the naming of TracingChannel it makes sense to call it a trace. If an expression of a collection of these units is described in this doc somewhere we could just call it a sub-trace. Many APMs think of it that way anyway, where there's not really a distinction between spans and traces, they're all just nested elements.

Reply:
Yes, using (internal) terms from APM vendors will not help here.
I thought using Span would be quite good because it's also used in OpenTelemetry which is quite popular meanwhile and open.
At some other places you used `single traceable action`. Matches better but is quite long.

Anyhow, feel free to keep it as it is. APM aware people will be likely able to understand this and others are likely lost at more places.

Reply:
The `single traceable action` terminology might be better here. It _is_ long, but it's also unambiguous and not so much based on loaded terminology like "span" or whatever else an APM might know the concept as. What do you think of changing it to that?

Reply:
sounds good.